pkg/converter/convert_test.go

```diff
@@ -1042,6 +1044,26 @@ func TestConvert(t *testing.T) {
 			},
 			command: execName,
 		},
+		{
+			name:        "convert with context specified, auth info not specified by the context should not be changed",
+			contextName: clusterName1,
+			authProviderConfig: map[string]string{
+				cfgEnvironment: envName,
+				cfgApiserverID: serverID,
+				cfgClientID:    clientID,
+				cfgTenantID:    tenantID,
+				cfgConfigMode:  "0",
+			},
+			overrideFlags: map[string]string{
+				flagLoginMethod: token.MSILogin,
+				"context":       clusterName1,
```

Comment:
nit: this flag name is repeating multiple times, consider adding a `flagContext` variable for this usage?